mongodb/vibe/db/mongo/connection.d

```diff
@@ -181,6 +183,56 @@ final class MongoConnection {
 			throw new MongoDriverException(format("Failed to connect to MongoDB server at %s:%s.", m_settings.hosts[0].name, m_settings.hosts[0].port), __FILE__, __LINE__, e);
 		}
 
+		if (m_settings.nextgen) {
+			import os = std.system;
+			import compiler = std.compiler;
+			string platform = compiler.name ~ " "
+				~ compiler.version_major.to!string ~ "." ~ compiler.version_minor.to!string;
+			// TODO: add support for os.version
+
+			version (X86_64)
+				string arch = "x86_64 ";
+			else version (X86)
+				string arch = "x86 ";
+			else version (AArch64)
+				string arch = "aarch64 ";
+			else version (ARM_HardFloat)
+				string arch = "armhf ";
+			else version (ARM)
+				string arch = "arm ";
+			else version (PPC64)
+				string arch = "ppc64 ";
+			else version (PPC)
+				string arch = "ppc ";
+			else
+				string arch = "unknown ";
+
+			Bson handshake = Bson.emptyObject;
+			handshake["isMaster"] = Bson(1);
+			handshake["client"] = Bson([
+				"driver": Bson(["name": Bson("vibe.db.mongo"), "version": Bson("0.8.4")]),
```

Comment:
Oh hard-coding the Vibe.d version might not be a good idea.
How about doing sth. like we did in D-Scanner?

https://github.com/dlang-community/D-Scanner/blob/master/dub.json

Reply:
eh maybe separate version than the vibe.d version would be a better idea because the driver doesn't update with every release

Reply:
For now, you could also use `vibe.core.core.vibeVersionString` - that will typically be the vibe-core version nowadays, but at least it won't get stale like a static string. Once the mongo driver gets separated out into its own repository, we can start an own version scheme.